python/cudf/cudf/core/dataframe.py

```diff
@@ -2997,10 +2998,10 @@ def agg(self, aggs, axis=None):
                             f"'Series' object"
                         )
                     result[key] = getattr(col, value)()
-            elif all([isinstance(val, Iterable) for val in aggs.values()]):
+            elif all([isinstance(val, abc.Iterable) for val in aggs.values()]):
                 idxs = set()
                 for val in aggs.values():
-                    if isinstance(val, Iterable):
+                    if isinstance(val, abc.Iterable):
                         idxs.update(val)
                     elif isinstance(val, str):
                         idxs.add(val)
```

Comment:
The order of these statements may be a bug. Strings meet the requirements of `Iterable`, so the string code path may never be taken here. Strings should probably be handled first. I'll fix this.